lib/mux.js

```diff
@@ -74,6 +77,8 @@ Mux.prototype._readIncoming = function() {
         if (chunk !== null) {
           accepting = out.write(chunk);
           streams.push(s);
+        } else {
+          s.emit('inbound_empty');
```

Comment:
Hmm in this case, is it the right place to emit `inbound_empty`? Shouldn't it be after the `while` completes?

Reply:
No I believe this is correct.
When a specific stream returns null on `s.read()` it means that that stream has no inbound left. That's the moment we want to emit the event.

If we emit the event when the while loop ends, there are two issues:

1. Which stream do we emit the event on? The stream being worked on is assigned here: `while (accepting && (s = streams.shift()))` and on exit we will have lost that reference.
2. If the loop ends because rabbit's writable stream stops accepting (it needs to drain), why should we emit `inbound_empty`? There is still inbound, it just needs to wait some time before the mux continues to write

Reply:
Oh I see! Makes sense